common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,33 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
+    StringBuilder textSB = new StringBuilder();
+
+    char outsideQuote = whichQuote(text);
+    if (outsideQuote != 0) {
+      char currentChar;
+      char nextChar;
+
+      for (int chIndex = 1; chIndex < text.length() - 1; chIndex++) {
+        currentChar = text.charAt(chIndex);
+        nextChar = text.charAt(chIndex + 1);
```

Comment:
what if nextChar is the closing quote?

Reply:
Also, we should have a unit test for this :)

Reply:
Not a concern because all that happens is `chindex++`.